### PR TITLE
Attach release binaries to every created release

### LINKED ISSUE #88: Attach binaries as part of releases
I am interested in publishing a (binary) package to AUR for this tool. To make this easier, it would be nice if (at least) a Linux build would be attached to each release.

Would you be open to adding a GitHub workflow that attaches a binary to each release? I am happy to submit a PR for that.

### PR DESCRIPTION
**Pull request by thomaseizinger**
Resolves #88.

Here is a test release: https://github.com/thomaseizinger/cargo-msrv/releases/tag/v0.7.3
Here is the workflow run that created it: https://github.com/thomaseizinger/cargo-msrv/actions/runs/1066616782

It is triggered on making a new release via Github.

### CODE REVIEW

File: .github/workflows/build-release-binaries.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      - id: create-archive-name
+        shell: python # Use python to have a prettier name for the archive on Windows.
+        run: |
+          import platform
+          os_info = platform.uname()
+
+          arch = os_info.machine
+
+          triple = "${{ matrix.target }}".split("-")
+          arch = triple[0]
```
Comment (foresterre) [inline .github/workflows/build-release-binaries.yaml:47]: Does `arch` here always overwrite the `arch` on line 44?

**Review by foresterre — APPROVED** [review]
Thanks for the PR! I'm happy with it, thanks!

I have left one minor comment, for my understanding on whether the initialisation of `arch` is necessary or not.

File: .github/workflows/build-release-binaries.yaml (hunk 1 above)
Comment (thomaseizinger) [reply .github/workflows/build-release-binaries.yaml:47]: Uh yeah, I think you are right!
Apologies, I copied this from one of our projects :)

Will fix!

**Comment by thomaseizinger** [comment]
Action run for new version: https://github.com/thomaseizinger/cargo-msrv/actions/runs/1077897661
Release: https://github.com/thomaseizinger/cargo-msrv/releases/tag/v0.7.4

**Comment by foresterre** [comment]
Great thanks!

**Comment by thomaseizinger** [comment]
> Build failed:
> 
>     * [clippy](https://github.com/foresterre/cargo-msrv/runs/3196964175)

Is this something I need to take care of @foresterre ?

**Comment by foresterre** [comment]
> 
> 
> > Build failed:
> > ```
> > * [clippy](https://github.com/foresterre/cargo-msrv/runs/3196964175)
> > ```
> 
> Is this something I need to take care of @foresterre ?

No, I will fix it in a few minutes

**Comment by foresterre** [comment]
bors retry

**Comment by foresterre** [comment]
Thanks! 😄

**Pull request merged**